gameconfig.lua

```diff
@@ -0,0 +1,313 @@
+-- Craft ingredients
+
+
+bm = {}
+-- MineClone2 support
+local mod_mcl_core = minetest.get_modpath("mcl_core") ~= nil
+if mod_mcl_core then
+	bm.gameconfig_dirt = "mcl_core:dirt"
+	bm.gameconfig_sand = "mcl_core:sand"
+	bm.gameconfig_gravel = "mcl_core:gravel"
+	bm.gameconfig_steel_ingot = "mcl_core:iron_ingot"
+	bm.gameconfig_gold_ingot = "mcl_core:gold_ingot"
+	bm.gameconfig_mese_crystal_fragment = "mesecons:redstone"
+	bm.gameconfig_torch = "mcl_torches:torch"
+	bm.gameconfig_diamond = "mcl_core:diamond"
+	bm.gameconfig_clay_lump = "mcl_core:clay_lump"
+	bm.gameconfig_water_bucket = "mcl_buckets:bucket_water"
+	bm.gameconfig_empty_bucket = "mcl_buckets:bucket_empty"
+	bm.gameconfig_dye_dark_grey = "mcl_dye:dark_grey"
+	-- No current equivalent so default to iron ingot instead.
+	bm.gameconfig_copper_ingot = "mcl_core:iron_ingot"
+	bm.gameconfig_tin_ingot = "mcl_core:iron_ingot"
+	bm.gameconfig_silver_ingot = "mcl_core:iron_ingot"
+else --fallback to Minetest Game's items.
+	bm.gameconfig_dirt = "default:dirt"
+	bm.gameconfig_sand = "default:sand"
+	bm.gameconfig_gravel = "default:gravel"
+	bm.gameconfig_copper_ingot = "default:copper_ingot"
+	bm.gameconfig_steel_ingot = "default:steel_ingot"
+	bm.gameconfig_gold_ingot = "default:gold_ingot"
+	bm.gameconfig_tin_ingot = "default:tin_ingot"
+	bm.gameconfig_mese_crystal_fragment = "default:mese_crystal_fragment"
+	bm.gameconfig_torch = "default:torch"
+	bm.gameconfig_diamond = "default:diamond"
+	bm.gameconfig_clay_lump = "default:clay_lump"
+	bm.gameconfig_water_bucket = "bucket:bucket_water"
+	bm.gameconfig_empty_bucket = "bucket:bucket_empty"
+	bm.gameconfig_dye_dark_grey = "dye:dark_grey"
+	-- No current equivalent so default to copper ingot instead.
+	bm.gameconfig_silver_ingot = "default:copper_ingot"
+end
+
+
+-- Sounds
```

Comment:
Everything below this line is almost sure bad idea, you already mentioned "support for new hypothetical game" but it is not just hypothetical as there's open issue in sound_api repo for that.

Duplicating code here means harder to maintain and more places for mistakes, using library means that you'll only need to update library which is then used by multiple different projects. Fixes and feature updates can be used everywhere without rewriting code for every mod that is using library.

If submodule requirement causes problems then there's other solutions for that but for these issues it would be better to open separate issues and pull request. Duplicating code and dropping central repository unnecessarily like that should not be accepted without very good reason and so far there does not seem to be actual reasons provided.

I'm not sure if you've changed something while copying code from sound api but if you did then those changes should almost sure go into sound api itself.

> Honestly, if it were purely up to me. I'd just add 2 sound files. The sounds are for 3 nodes that originate here in basic_materials.
There's really no reason to have such an involved system for these nodes.

You probably also understand that doing it this way would take away game specific experience while increase media footprint, especially if every mod that needs just few sounds would follow this logic?

----

It is true that sound api as a submodule has potential for additional problems but ignoring upsides of library and purpose of library wont bring better solutions. Basically some discussion is needed if mentioning that _"There's really no reason to have such an involved system for these nodes."_ and at the same time copy pasting a lot of code that duplicates maintenance requirements effectively creating two such an involved systems.

If you see some problem in sound api then best to open separate issue for sound api.
If you think sound api is not useful enough for basic materials then open separate issue to discuss that, prefereably include some reasoning why something else would be better and/or why library is worse.